mitodl/mitxonline
# add migration to delete duplicate discount redemption

# What are the relevant tickets?




https://github.com/mitodl/hq/issues/1765

# Description (What does it do?)

Adds a migration to delete the duplicate discount redemptions on production https://bi.odl.mit.edu/queries/1293.
It keeps the last DiscountRedemption per order.

https://github.com/mitodl/mitxonline/pull/1721 has fixed the logic, but we also need to fix the data. 


# How can this be tested?

- Create orders
- Create duplicate DiscountRedemption for same orders either from shell or from admin. 
   - To add from admin, you will need to comment out `readonly_fields` in `DiscountRedemptionAdmin`
- Run migration
- Verify from http://mitxonline.odl.local:8013/admin/ecommerce/discountredemption/ . There should be only one redemption per order. For duplicates, it should keep the most recent one.

Before Migration
![image](https://github.com/mitodl/mitxonline/assets/3138890/0a48da6c-275f-4c69-8d85-a5ca6c7f351b)

After migration
<img width="1370" alt="image" src="https://github.com/mitodl/mitxonline/assets/3138890/2f3aa38e-c71f-49f4-b4fb-7f8a07ba3d3d">
